src/utils.rs

```diff
@@ -115,4 +134,40 @@ mod tests {
             assert_line_written("foo bar\tbaz", true, 8, "foo bar baz");
         }
     }
+
+    mod modification_time {
+        use super::*;
+
+        #[test]
+        fn set_time() {
+            use chrono::{DateTime, Local};
+            use std::fs::File;
+            use std::time::SystemTime;
+
+            let target = "target/utils";
+            let _ = std::fs::create_dir(target);
+            let filename = &format!("{target}/foo");
+            let temp = File::create(filename).unwrap();
```

Comment:
Can you use a [tempfile.NamedTempFile](https://docs.rs/tempfile/3.10.1/tempfile/struct.NamedTempFile.html) instead?

Reply:
I don't think there are APIs to set modification time for a `NamedTempFile`.

Reply:
`NamedTempFile::as_file()` exposes a reference to the underlying file, so you could do something like this:

```rust
use tempfile::NamedTempFile;
let temp = NamedTempFile::new().unwrap();
let current = SystemTime::now();
let _ = temp.as_file().set_modified(current);
…
assert_eq!(current, get_modification_time(&temp.path().to_string_lossy()));
```